configuration/configuration.go

```diff
@@ -685,17 +717,39 @@ func (c *ConfigurationData) GetCacheControlUser() string {
 }
 
 // GetDeprecatedServiceAccountPrivateKey returns the deprecated service account private key (if any) and its ID
-// that is used to verify the service account authentication tokens during key rotation.
+// that is used to verify service account authentication tokens during key rotation.
 func (c *ConfigurationData) GetDeprecatedServiceAccountPrivateKey() ([]byte, string) {
 	return []byte(c.v.GetString(varServiceAccountPrivateKeyDeprecated)), c.v.GetString(varServiceAccountPrivateKeyIDDeprecated)
 }
 
 // GetServiceAccountPrivateKey returns the service account private key and its ID
-// that is used to sign the service account authentication tokens.
+// that is used to sign service account authentication tokens.
 func (c *ConfigurationData) GetServiceAccountPrivateKey() ([]byte, string) {
 	return []byte(c.v.GetString(varServiceAccountPrivateKey)), c.v.GetString(varServiceAccountPrivateKeyID)
 }
 
+// GetDeprecatedUserAccountPrivateKey returns the deprecated user account private key (if any) and its ID
+// that is used to verify user access and refresh tokens during key rotation.
+func (c *ConfigurationData) GetDeprecatedUserAccountPrivateKey() ([]byte, string) {
+	return []byte(c.v.GetString(varUserAccountPrivateKeyDeprecated)), c.v.GetString(varUserAccountPrivateKeyIDDeprecated)
+}
+
+// GetUserAccountPrivateKey returns the service account private key and its ID
+// that is used to sign user access and refresh tokens.
+func (c *ConfigurationData) GetUserAccountPrivateKey() ([]byte, string) {
+	return []byte(c.v.GetString(varUserAccountPrivateKey)), c.v.GetString(varUserAccountPrivateKeyID)
+}
+
+// GetDevModePublicKey returns additional public key and its ID which should be used by the Auth service in Dev Mode
+// For example a public key from Keycloak
+// Returns false if in in Dev Mode
+func (c *ConfigurationData) GetDevModePublicKey() (bool, []byte, string) {
+	if c.IsPostgresDeveloperModeEnabled() {
+		return true, []byte(devModePublicKey), devModePublicKeyID
```

Comment:
Is the intention of this method to ensure we add the existing public keys used be prod-preview.. and enable people to stay logged in locally in dev mode ?

In that case, the public keys should be from 
https://sso.prod-preview.openshift.io/auth/realms/fabric8/protocol/openid-connect/certs
or
https://sso.prod-preview.openshift.io/auth/realms/fabric8-test/protocol/openid-connect/certs  ?

Reply:
> Is the intention of this method to ensure we add the existing public keys used by prod-preview.. and enable people to stay logged in locally in dev mode ?

Yeah.

> In that case, the public keys should be from...

In dev mode we use fabric8-test. So this key is from https://sso.prod-preview.openshift.io/auth/realms/fabric8-test/protocol/openid-connect/certs
Please note that cert endpoint returns keys in JWK format. Not in PEM (as in our configuration).
You can see the PEM key here - https://sso.prod-preview.openshift.io/auth/realms/fabric8-test

Reply:
Thanks. @alexeykazakov , Is the `DefaultUserAccountPrivateKey` from a freshly generated keypair ? When run locally, this is the one that would be used to sign the tokens, right ? In that case, would local auth be  able to talk to keycloak for spaces' related authorization ?

Reply:
Yes, it's freshly generated keys. And no it won't be able to talk to Keycloak for space authZ. So, basically it's tricky to use collaboration API in dev mode now.
WIT space authZ should be fine because AuthZ is disabled in WIT in Dev Mode by default.